### PR TITLE
Workflow change project

### PR DESCRIPTION
**Pull request by balsdorf**
This can be used to map values for fields between projects or do other custom changes.

### CODE REVIEW

**Comment by glensc** [comment]
> Should this be called on the workflow of the old project or the new project?

if you're not sure how, you could add two methods:
- movedFromProject
- movedToProject

so you can choose which one to implement.

**Comment by balsdorf** [comment]
Good idea. I've done that now. I'll do a bit more testing before merge.

Since ChangeLog is conflicting now, should I rebase once everything is good to go?

**Comment by glensc** [comment]
i'm planning to make release shortly, so it will conflict even more. and this PR should be merged after the release, to give it more testing ground.

so yes, rebase after release made (and fix `@since` tags)

File: lib/eventum/class.workflow.php [hunk 1]
```diff
@@ -930,19 +930,37 @@ public static function cryptoDowngradeConfig($prj_id = 1)
...
+    }
+
+    /**
+     * Called when an issue is moved to this project from another.
+     *
+     * @param $prj_id
+     * @param $issue_id
+     * @param $old_prj_id
      * @since 3.1.6
+     * @return null
```
Comment (glensc) [inline lib/eventum/class.workflow.php]: it doesn't `return` anything. remove this line

File: lib/eventum/workflow/class.abstract_workflow_backend.php [hunk 2]
```diff
@@ -685,22 +685,34 @@ public function cryptoDowngradeConfig()
...
+    }
+
+    /**
+     * Called when an issue is moved to this project from another.
+     *
+     * @param $prj_id
+     * @param $issue_id
+     * @param $old_prj_id
+     * @since 3.1.6
+     */
```
Comment (glensc) [inline lib/eventum/workflow/class.abstract_workflow_backend.php]: as minimum, should fill param types

**Comment by glensc** [comment]
relased, you may rebase now

**Comment by balsdorf** [comment]
Well I royally made a mess with git but I think I managed to fix it all.

**Comment by glensc** [comment]
you can see the Commits tab (https://github.com/eventum/eventum/pull/223/commits) what are the actual commits associated, looks good right now.

**Pull request merged**